### PR TITLE
Option to make password visible when entering

### PR DESCRIPTION
**Pull request by robert00s**
Reference: #2882 
It's now possible to visible password when entering (connect to the wifi). After click in "Swich visible" you can show/hide password. 

screenshot 2017-08-01 17-27-32

screenshot 2017-08-01 17-28-34

I have no way of checking (on the emulator and the Kindle there is no option to add a wifi network), so ask to see if it works.

### CODE REVIEW

**Comment by Frenzie** [comment]
I wonder if it might make more sense to add a new property like `show_toggle` which would then display a button or checkmark like that underneath the actual input dialog.

Then you'd get

```lua
if self.show_toggle then
-- etc
```

We could then even opt to default to true and it'd take care of all password dialogs in one fell swoop.

Does that make sense to you?

**Comment by robert00s** [comment]
Good catch. This is better solution.

File: frontend/ui/widget/inputdialog.lua [hunk 1]
```diff
@@ -72,6 +72,7 @@ local InputDialog = InputContainer:new{
     buttons = nil,
     input_type = nil,
     enter_callback = nil,
+    show_toggle = nil,
```
Comment (Frenzie) [inline frontend/ui/widget/inputdialog.lua]: Looking at it out of context I think `show_password_toggle` would be clearer.

File: frontend/ui/widget/inputdialog.lua [hunk 2]
```diff
@@ -148,6 +149,23 @@ function InputDialog:init()
         parent = self,
     }
 
+    if self.show_toggle then
+        local button_switch = {
+            {
+                text = ("Switch visible"),
```
Comment (Frenzie) [inline frontend/ui/widget/inputdialog.lua]: I propose something like `Show password` instead.

File: frontend/ui/widget/inputdialog.lua [hunk 3]
```diff
@@ -148,6 +149,23 @@ function InputDialog:init()
...
+                callback = function()
+                    if self._input_widget.text_type == "text" then
+                        self._input_widget.text_type = "password"
+                        self:changeTextType("password")
+                    else
+                        self._input_widget.text_type = "text"
```
Comment (Frenzie) [inline frontend/ui/widget/inputdialog.lua]: More specifically, I propose to change `.text` like this

`☐ Show password` (hidden)
`√ Show password` (shown)

(Alternatively you could change it to `Show password` but I think that might be a touch disorienting)

**Comment by Frenzie** [comment]
Any thoughts on enabling it by default?

**Comment by robert00s** [comment]
+ Enabled by defalut
+ show_password_toggle
+ Show password

Switching between `☐ Show password (hidden)` and `√ Show password (shown)` is difficult to do. I don't know how to change button text after click on it in inputdialog widget. I propose to leave text "Show password".

**Pull request merged**